src/Injector/AbstractInjector.php

```diff
@@ -162,6 +181,155 @@ public function inject($package, $type, IOInterface $io)
     }
 
     /**
+     * Inject component $package after all dependencies into $config and return true.
+     * If any of dependencies is not registered the method will write error to $io
+     * and will also return true, to prevent injecting this package later.
+     * Method return false only in case when dependencies for the package are not found.
+     *
+     * @param string $package
+     * @param string $config
+     * @param IOInterface $io
+     * @return bool
+     */
+    private function injectAfterDependencies($package, $config, IOInterface $io)
+    {
+        $moduleClass = sprintf('%s\\%s', $package, 'Module');
+        if (class_exists($moduleClass) && method_exists($moduleClass, 'getModuleDependencies')) {
+            $module = new $moduleClass();
+            $dependencies = $module->getModuleDependencies();
+
+            if ($dependencies) {
+                foreach ($dependencies as $dependency) {
+                    if (! $this->isRegisteredInConfig($dependency, $config)) {
+                        $io->write(sprintf(
+                            '<error>    Dependency %s is not registered in the configuration</error>',
+                            $dependency
+                        ));
+
+                        return true;
```

Comment:
Hm... I think I wasn't clear enough in my comment.
Please have a look on comment before method in php-docs.
I don't like this `true` because in this case the package also is **not** injected, but this method return `true`, to prevent injecting it in calling method. It also emit error - please see lines above - 204-207.